pkg/syncer/metadatasyncer_test.go

```diff
@@ -760,3 +765,758 @@ func TestCalculateVMServiceStoragePolicyUsageReservedForNamespace(t *testing.T)
+			},
+			setupMetadataSyncer: func(t *testing.T) *metadataSyncInformer {
+				// Create a fake k8s client and informer factory to get listers
+				// Note: We create a new informer factory directly instead of using
+				// the singleton InformerManager to avoid interference with other tests.
+				// The listers work fine without starting the informers.
+				k8sClient := testclient.NewSimpleClientset()
+				informerFactory := informers.NewSharedInformerFactory(k8sClient, 0)
+				return &metadataSyncInformer{
+					pvLister:  informerFactory.Core().V1().PersistentVolumes().Lister(),
+					pvcLister: informerFactory.Core().V1().PersistentVolumeClaims().Lister(),
+				}
+			},
+			expectError: false,
+		},
+		{
+			name: "Negative value update skipped - StoragePolicyUsage with negative reserved",
+			setupClient: func() client.Client {
+				scheme := runtime.NewScheme()
+				_ = cnsoperatorv1alpha1.AddToScheme(scheme)
+				_ = sqperiodicsyncv1alpha1.AddToScheme(scheme)
+
+				spq := &storagepolicyv1alpha2.StoragePolicyQuota{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "spq-1",
+						Namespace: "test-ns",
+					},
+				}
+
+				// Create StoragePolicyUsage with negative reserved value
+				negTenGi := resource.MustParse("-10Gi")
+				spu := &storagepolicyv1alpha2.StoragePolicyUsage{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "spu-1",
+						Namespace: "test-ns",
+					},
+					Spec: storagepolicyv1alpha2.StoragePolicyUsageSpec{
+						StoragePolicyId:       "policy-1",
+						ResourceExtensionName: VMExtensionServiceName,
+					},
+					Status: storagepolicyv1alpha2.StoragePolicyUsageStatus{
+						ResourceTypeLevelQuotaUsage: &storagepolicyv1alpha2.QuotaUsageDetails{
+							Reserved: &negTenGi,
+						},
+					},
+				}
+
+				sqPeriodicSync := &sqperiodicsyncv1alpha1.StorageQuotaPeriodicSync{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      StorageQuotaPeriodicSyncInstanceName,
+						Namespace: "vmware-system-csi",
+					},
+				}
+
+				return fake.NewClientBuilder().
+					WithScheme(scheme).
+					WithObjects(spq, spu, sqPeriodicSync).
+					Build()
+			},
+			setupMetadataSyncer: func(t *testing.T) *metadataSyncInformer {
+				// Save original function
+				origFetchStorageClassToStoragePolicyMapping := fetchStorageClassToStoragePolicyMapping
+				t.Cleanup(func() {
+					fetchStorageClassToStoragePolicyMapping = origFetchStorageClassToStoragePolicyMapping
+				})
+				// Mock fetchStorageClassToStoragePolicyMapping
+				fetchStorageClassToStoragePolicyMapping = fakeFetchStorageClassToStoragePolicyMapping
+
+				k8sClient := testclient.NewSimpleClientset()
+				informerFactory := informers.NewSharedInformerFactory(k8sClient, 0)
+				return &metadataSyncInformer{
+					pvLister:  informerFactory.Core().V1().PersistentVolumes().Lister(),
+					pvcLister: informerFactory.Core().V1().PersistentVolumeClaims().Lister(),
+				}
+			},
+			expectError: false, // Function should skip negative values and continue
```

Comment:
ideally we should check for sqPeriodicSync to not have negative value populated as validation of this test, can be taken up in next PR